spark/src/main/scala/com/cloudera/livy/spark/client/ClientSession.scala

```diff
@@ -54,14 +69,17 @@ class ClientSession(val sessionId: Int, createRequest: CreateClientRequest) exte
   }
 
   def addFile(uri: URI): Unit = {
-    getClient.foreach(_.addFile(uri))
+    _lastActivity = System.currentTimeMillis()
+    client.addFile(uri)
   }
 
   def addJar(uri: URI): Unit = {
-    getClient.foreach(_.addJar(uri))
+    _lastActivity = System.currentTimeMillis()
```

Comment:
This would probably be cleaner if the parent class instead had a `recordActivity` method.